src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -543,7 +543,14 @@ await EnqueueWorkItemAsync(projectChanges.OldProject.GetDocument(changedDocument
             private async Task EnqueueWorkItemAsync(Document oldDocument, Document newDocument)
             {
                 var differenceService = newDocument.GetLanguageService<IDocumentDifferenceService>();
-                if (differenceService != null)
+
+                if (differenceService == null)
+                {
+                    // For languages that don't use a Roslyn syntax tree, they don't export a document difference service.
+                    // The whole document should be considered as changed in that case.
+                    await EnqueueWorkItemAsync(newDocument, InvocationReasons.DocumentChanged).ConfigureAwait(false);
```

Comment:
actually, I think it should be SyntaxChanged to make its cost to be low. if any one want to have expansive propagation, they should export IDocumentDifferenceService so that they can control its cost.